implementations/rust/ockam/ockam_identity/src/secure_channel/decryptor.rs

```diff
@@ -1,39 +1,137 @@
 use crate::identity::IdentityError;
-use crate::secure_channel::encryptor::Encryptor;
+use crate::secure_channel::encryptor::{Encryptor, KEY_RENEWAL_INTERVAL};
 use ockam_core::compat::sync::Arc;
 use ockam_core::compat::vec::Vec;
-use ockam_core::vault::{KeyId, SymmetricVault};
+use ockam_core::vault::KeyId;
 use ockam_core::Result;
+use ockam_key_exchange_xx::XXInitializedVault;
+use tracing::warn;
 
-#[derive(Clone)]
 pub(crate) struct Decryptor {
-    key: KeyId,
-    vault: Arc<dyn SymmetricVault>,
+    current_key: KeyId,
+    current_key_nonce: u64,
+
+    previous_key: Option<KeyId>,
+    vault: Arc<dyn XXInitializedVault>,
+    nonce_tracker: NonceTracker,
+}
+
+pub(crate) struct NonceTracker {
```

Comment:
Could you please move it either to the separate file or closer to the implementation?